src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -93,6 +155,7 @@ protected override void Dispose(bool disposing)
             state.CancellationToken = cancellationToken;
             state.RequestMessage = request;
             state.Handler = this;
+            state.ProxyPolicy = _proxyPolicy;
```

Comment:
This code is still wrong regarding when to return FALSE for `UseProxy`:

`if (_proxyPolicy == CurlProxyUsePolicy.UseDefaultProxy)`